azurerm/internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -707,3 +720,43 @@ func resourceKubernetesClusterNodePoolDelete(d *schema.ResourceData, meta interf
 
 	return nil
 }
+
+func upgradeSettingsSchema() *schema.Schema {
+	return &schema.Schema{
+		Type:     schema.TypeList,
+		Optional: true,
+		MaxItems: 1,
+		Elem: &schema.Resource{
+			Schema: map[string]*schema.Schema{
+				"max_surge": {
+					Type:     schema.TypeString,
+					Optional: true,
+				},
+			},
+		},
+	}
+}
+
+func expandUpgradeSettings(input []interface{}) *containerservice.AgentPoolUpgradeSettings {
+	if len(input) == 0 {
+		return nil
+	}
+	upgradeSettingInput := input[0].(map[string]interface{})
+	upgradeSetting := containerservice.AgentPoolUpgradeSettings{}
+
+	if maxSurgeRaw := upgradeSettingInput["max_surge"].(string); maxSurgeRaw != "" {
+		upgradeSetting.MaxSurge = utils.String(maxSurgeRaw)
+	}
+	return &upgradeSetting
```

Comment:
meaning we could likely reduce this too:

```suggestion
	setting := &containerservice.AgentPoolUpgradeSettings{}
	if len(input) == 0 {
		return setting
	}

	v := input[0].(map[string]interface{})
	if maxSurgeRaw := v["max_surge"].(string); maxSurgeRaw != "" {
		setting.MaxSurge = utils.String(maxSurgeRaw)
	}
	return setting
```